skins/laika/src/components/pages/UpdateAddress.vue

```diff
@@ -0,0 +1,155 @@
+import { action } from '@/store/util';
+import { AddressTypeModel } from "@/view_models/AddressTypeModel";
+import { mapGetters } from "vuex";
+
+export default Vue.extend( {
+	name: 'update-address',
+	components: {
+		Name,
+		Postal,
+	},
+	beforeMount() {
+		this.setAddressType( this.isCompany ? AddressTypeModel.COMPANY : AddressTypeModel.PERSON );
+	},
+	data: function (): { formData: AddressFormData } {
+		return {
+			formData: {
+				salutation: {
+					name: 'salutation',
+					value: '',
+					pattern: '^(Herr|Frau)$',
+					optionalField: false,
+				},
+				title: {
+					name: 'title',
+					value: '',
+					pattern: '',
+					optionalField: true,
+				},
+				companyName: {
+					name: 'companyName',
+					value: '',
+					pattern: '^.+$',
+					optionalField: false,
+				},
+				firstName: {
+					name: 'firstName',
+					value: '',
+					pattern: '^.+$',
+					optionalField: false,
+				},
+				lastName: {
+					name: 'lastName',
+					value: '',
+					pattern: '^.+$',
+					optionalField: false,
+				},
+				street: {
+					name: 'street',
+					value: '',
+					pattern: '^.+$',
+					optionalField: false,
+				},
+				city: {
+					name: 'city',
+					value: '',
+					pattern: '^.+$',
+					optionalField: false,
+				},
+				postcode: {
+					name: 'postcode',
+					value: '',
+					pattern: '^[0-9]{4,5}$',
+					optionalField: false,
+				},
+				country: {
+					name: 'country',
+					value: 'DE',
+					pattern: '',
+					optionalField: false,
+				},
+			},
+		};
+	},
+	props: {
+		updateAddressURL: String,
+		isCompany: Boolean,
+		countries: Array as () => Array<String>,
+	},
+	computed: {
+		fieldErrors: {
+			get: function (): AddressValidity {
+				return Object.keys( this.formData ).reduce( ( validity: AddressValidity, fieldName: string ) => {
+					if( !this.formData[ fieldName ].optionalField ) {
+						validity[ fieldName ] = this.$store.state.address.validity[ fieldName ] === Validity.INVALID;
+					}
+					return validity;
+				}, ( {} as AddressValidity ) );
+			},
+		},
+		...mapGetters( NS_ADDRESS, [
+			'addressType',
+		] ),
+	},
+	methods: {
+		validateForm(): Promise<ValidationResult> {
+			return this.$store.dispatch( action( NS_ADDRESS, validateAddress ), this.$props.validateAddressUrl );
+		},
+		onFieldChange( fieldName: string ): void {
+			this.$store.dispatch( action( NS_ADDRESS, setAddressField ), this.$data.formData[ fieldName ] );
+		},
+		setReceiptOptedOut( optedOut: boolean ): void {
+			this.$store.dispatch( action( NS_ADDRESS, setReceiptOptOut ), optedOut );
+		},
+		setAddressType( addressType: AddressTypeModel ): void {
+			this.$store.dispatch( action( NS_ADDRESS, setAddressType ), addressType );
+		},
+		submit() {
+			this.validateForm().then( ( validationResult: ValidationResult ) => {
+				console.log( validationResult );
+				if ( validationResult.status === 'OK' ) {
+					const form = this.$refs.form as HTMLFormElement;
+					form.submit();
```

Comment:
If we want to track this form, it should happen before the submit. 
`trackFormSubmission( form );`